File: base.mvp/src/main/java/alvin/base/mvp/MainActivity.java
package alvin.base.mvp;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Button;

import alvin.base.mvp.basic.views.BasicMainActivity;
import alvin.base.mvp.scope.views.ScopeActivity;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ButterKnife.bind(this);
    }

    @OnClick({
            R.id.btn_basic,
            R.id.btn_scope
    })
    public void onButtonsClick(Button b) {
        Intent intent;

        switch (b.getId()) {
        case R.id.btn_basic:
            intent = new Intent(this, BasicMainActivity.class);
            break;
        case R.id.btn_scope:
            intent = new Intent(this, ScopeActivity.class);
            break;
        default:
            intent = null;
        }

        if (intent != null) {
            startActivity(intent);
        }
    }
}

File: ui.list/src/main/java/alvin/list/MainActivity.kt
package alvin.list

import alvin.list.list.views.ListMainActivity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import butterknife.ButterKnife
import butterknife.OnClick

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        ButterKnife.bind(this)
    }

    @OnClick(R.id.btn_list, R.id.btn_grid)
    fun onButtonClick(b: Button) {
        var intent: Intent? = null

        when (b.id) {
            R.id.btn_list -> {
                intent = Intent(this, ListMainActivity::class.java)
            }
            R.id.btn_grid -> {
                intent = null
            }
        }

        if (intent != null) {
            startActivity(intent)
        }
    }
}
